mempool/mempool.go

```diff
@@ -337,27 +352,43 @@ func (mem *Mempool) CheckTx(tx types.Tx, cb func(*abci.Response)) (err error) {
 	}
 	reqRes := mem.proxyAppConn.CheckTxAsync(tx)
 	if cb != nil {
-		reqRes.SetCallback(cb)
+		composedCallback := func(res *abci.Response) {
+			cb(res)
+			mem.reqResCb(tx, txInfo.PeerID)(res)
+		}
+		reqRes.SetCallback(composedCallback)
+	} else {
+		reqRes.SetCallback(mem.reqResCb(tx, txInfo.PeerID))
 	}
 
 	return nil
 }
 
 // ABCI callback function
-func (mem *Mempool) resCb(req *abci.Request, res *abci.Response) {
-	if mem.recheckCursor == nil {
-		mem.resCbNormal(req, res)
-	} else {
-		mem.metrics.RecheckTimes.Add(1)
-		mem.resCbRecheck(req, res)
+func (mem *Mempool) resCb() abcicli.Callback {
```

Comment:
Also, does it even make sense to have this function do anything? Since we're always setting the callback on the reqRes now, why not just handle the recheck login in there too?